cocos/2d/CCNode.h

```diff
@@ -1881,9 +1895,17 @@ class CC_DLL Node : public Ref
     mutable bool _additionalTransformDirty; ///< transform dirty ?
     bool _transformUpdated;         ///< Whether or not the Transform object was updated since the last frame
 
-    int _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
+    union {
+        struct {
+            std::int32_t z; // The original localZOrder
+            std::uint32_t a; // Order Of Arrival, for avoid sort problem with unstable_sort algorithm.
+        } detail;
+        std::int64_t value; // The value to be used in sort
+    } _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
```

Comment:
Yep, my analyzation of negative is wrong. But it still has problem with big endian and little endian.